### PR TITLE
Added some utilities that will be helpful for testing

### PR DESCRIPTION
**Pull request by TIHan**
This adds some utilities that will be useful for testing soon. It also enables preview SDKs to be used on CI builds.

### CODE REVIEW

File: tests/fsharp/test-framework.fs [hunk 1]
```diff
@@ -105,6 +105,9 @@ module Commands =
     let ildasm exec ildasmExe flags assembly =
         exec ildasmExe (sprintf "%s %s" flags (quotepath assembly))
 
+    let ilasm exec ilasmExe flags assembly =
```
Comment (KevinRansom) [inline tests/fsharp/test-framework.fs]: Is this for desktop only tests?

File: tests/fsharp/Compiler/Utilities.fs [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+open System.Collections.Immutable
+open Microsoft.CodeAnalysis
+open Microsoft.CodeAnalysis.CSharp
+
+// This file mimics how Roslyn handles their compilation references for compilation testing
+
+[<RequireQualifiedAccess>]
+type TargetFramework =
+    | NetStandard20
+    | NetCoreApp30
```
Comment (KevinRansom) [inline tests/fsharp/Compiler/Utilities.fs]: We might want add NetStandard21 as well.

Comment (TIHan) [reply tests/fsharp/Compiler/Utilities.fs]: Not yet. The test resources from Roslyn to make this work does not include NetStandard21 yet.

Comment (KevinRansom) [reply tests/fsharp/Compiler/Utilities.fs]: Ahhh .. that makes sense

File: tests/fsharp/test-framework.fs (hunk 1 above)
Comment (TIHan) [reply tests/fsharp/test-framework.fs]: It doesn't have to be.

**Pull request merged**